docker/build.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+set -e
+
+build() {
+    ACTION=${1}
+    PHP_VERSION=${2}
+
+    DOCKER_IMAGE_TAG="ghcr.io/zenstruck/foundry/php-${PHP_VERSION}:latest"
+
+    # From inside the containers, docker host ip is different in linux and macos
+    UNAME=$(uname -s)
+    if [ "${UNAME}" = "Linux" ]; then
+        XDEBUG_HOST="172.17.0.1"
+    elif [ "${UNAME}" = "Darwin" ]; then
+        XDEBUG_HOST="host.docker.internal"
+    fi
+
+    docker pull ${DOCKER_IMAGE_TAG} || true
+
+    DOCKER_BUILDKIT=1 docker buildx build \
+        --tag "${DOCKER_IMAGE_TAG}" \
+        --cache-from "${DOCKER_IMAGE_TAG}" \
```

Comment:
using `docker buildx` make the magic of docker cache: first, we pull the image, then we take cache from it. Then we benefit from docker cache layering AND we also build if Dockerfile changed